src/Distribution/Server/Util/Validators.hs

```diff
@@ -30,11 +30,13 @@ guardValidLookingEmail str = either errBadEmail return $ do
   guard (T.all (not.isAngle) str) ?! "Please use just the email address, not \"name\" <person@example.com> style."
   where
     isAngle c = c == '<' || c == '>'
-    hasAtSomewhere =
-      let (before, after) = T.span (/= '@') str
-       in T.length before >= 1
-       && T.length after  >  1
-       && not ('@' `T.elem` after)
+    hasAtSomewhere = case T.span (/= '@') str of
+      (before, rest)
+        | Just (_, after) <- T.uncons rest ->
+          T.length before >= 1
+            && T.length after > 0
```

Comment:
What is the significance of using `>= 1` vs. `> 0`?  Why aren't both checks written as `not . T.null`?

Reply:
Thanks, I didn't spot this.

`Text` is strict here so I don't think there's any functional difference. I didn't realise that `T.length` is O(n) so `not . T.null` is asymptotically better here, although we've already paid that cost with the guard that `T.length str <= 100` which would maybe be more efficient as `T.compareLength str 101 == LT`. We could consolidate the check further with ``((&&) `on` (not . T.null)) before after``.

Reply:
+1.

> We could consolidate the check further with `` ((&&) `on` (not . T.null)) before after ``.

Mmh, this last suggestion seems just to make the code less readable...

Reply:
I am following up on my suggestion in
- #1077